apps/studio/src/features/editing-experience/components/ComplexEditorStateDrawer.tsx

```diff
@@ -74,11 +88,36 @@ export default function ComplexEditorStateDrawer(): JSX.Element {
           onClick={() => {
             setDrawerState({ state: "root" })
             setSavedPageState(previewPageState)
+            mutate({
+              pageId,
+              siteId,
+              content: JSON.stringify({
+                ...pageContent,
+                content: previewPageState,
+              }),
```

Comment:
having to fetch page content just to stringify here isn't ideal and i'll refactor when all our "content updating" features are in